### PR TITLE
789 weechat time format

### PR DESCRIPTION
**Pull request by pmelanson**
WIP PR. I'll go over this sometime soon, but atm it seems to work, at least when I tested it with `weechat.look.buffer_time_format = %H:%M:%S` and `%I:%M:%S` ie 24h and 12h time.

I also touched the fetchConfValue code slightly, adding the ability to pass in a callback function. Let me know if my JS flailings are wrong there.

Addresses #789 

I'll send this to people who brought up the 12h/24h time in the channel.

### CODE REVIEW

**Comment by pmelanson** [comment]
This also means we'll be able to remove the timestamp-related settings from the settings page, just haven't gotten around to that

**Comment by pmelanson** [comment]
Currently, my approach is to try and detect whether Weechat's time format is one of (long|short) (12|24) hour time.

An alternate approach is to do a lot of replacing, for example if weechat's time format is called `weechatTimeFormat`,

`weechatTimeFormat.replace("%H", "HH") // 24h hour field`
`weechatTimeFormat.replace("%I", "hh") // 12h hour field`

Instead of our current approach, which just tries to slot the user's weechat configuration into one of the four time formats I kind of arbitrarily defined.

Thoughts on this alternate approach? Our current approach kind of aggressively slots users into one of only four different time formats, but all four are sane formats and we are sure to always slot them into one of these formats.

The alternate approach has the benefit of respecting the user's configuration, but introduces more complexity.

Thoughts?

**Comment by lorenzhs** [comment]
Wow, cool! I think it's better to keep the slot approach, as weechat time format strings can also contain colours and other stuff that would be super hard to parse. I don't quite understand what the `'` do (and I haven't actually tested it) but nice work!

**Comment by pmelanson** [comment]
Oh yeah, the `'` are probably the jankiest thing in here. I think I'm going to have to add a comment explaining them

So you'll notice that some of the strings in there, like `secondsFormat`, have `'`. The `'` are angular's way of delimiting string literals, but having two `'` in a row is a literal `'`. So my first approach was just to surround strings like `timeDelimiter` and `secondsFormat` in `'`, but then when I concatenated two such strings together there would be a stray `'` in the resulting time format. For example, people were reporting `23:23':23`.

TL;DR two `'` in `$rootScope.angularTimeFormat` will show up as a literal `'` in the final timestamp, so we have to be careful about making sure two `'` don't show up together. I'll try to not let it get in the way of the code clarity though, and also add comments to what I have now.

**Comment by lorenzhs** [comment]
Oh right the whole thing is dumped in the `$date` filter. Yeah now it makes sense, thanks.

**Comment by pmelanson** [comment]
Cleaned up the code, there's now no jankiness around `'` and also added comments. Made a new commit that removes the timestamp settings checkboxes.
